### PR TITLE
Fixes for dcp 707 and some refactoring

### PR DESCRIPTION
**Pull request by jacobwindsor**
- Fixes the bug Enrique found
- Does some refactoring
- Removes the pysradb library for some custom solution. See `get_srp_accession_from_geo`
  - I'm aware the list comprehensions are a little bit hard to reason about but a bunch of nested for loops was even worse

### CODE REVIEW

**Review by aaclan-ebi — APPROVED** [review]
Is it worth adding unit tests?

File: geo_to_hca/utils/sra_utils.py [hunk 1]
```diff
@@ -1,105 +1,124 @@
...
+
+
+def request_accession_info(accessions: [], accession_type: str) -> object:
+    """
+    Function which sends a request to NCBI SRA database to get an xml file with metadata about a
+    given list of biosample or experiment accessions. The xml contains various metadata fields.
+    """
+    if accession_type == 'biosample':
+        url = f'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch/fcgi?db=biosample&id={",".join(accessions)}'
+    if accession_type == 'experiment':
```
Comment (ke4) [inline geo_to_hca/utils/sra_utils.py:95]: What if `accession_type` is not `biosample` or `experiment`?
In that case `url` variable won't be initialised.

**Review by ke4 — APPROVED** [review]
I added a small comment, but otherwise LGTM

File: geo_to_hca/utils/sra_utils.py (hunk 1 above)
Comment (jacobwindsor) [reply geo_to_hca/utils/sra_utils.py:95]: good question... I don't know. I inherited this. I guess it's always one of the two as we never had an issue with this

**Comment by jacobwindsor** [comment]
@aaclan-ebi yes! I thought there were tests but actually they're in broker. They're sort of testing this so I will either move them or write some new ones

**Pull request merged**